ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/rest/RestTestExecutor.java

```diff
@@ -215,14 +217,19 @@ private void produceInputs(final Map<String, List<Record>> inputs) {
           topicInfo.getKeySerializer(),
           topicInfo.getValueSerializer()
       )) {
-        for (final Record record : records) {
-          producer.send(new ProducerRecord<>(
-              topicName,
-              null,
-              record.timestamp().orElse(0L),
-              record.key(),
-              record.value()
-          ));
```

Comment:
This code was neither waiting for the message to actually be sent, not failing if the message failed to send.

Reply:
How did the tests pass if the test wasn't waiting for the records to be sent? Was the downstream validation waiting for records to arrive? (Took a quick skim of the code but didn't see it.) Was it just luck that record production tended to be quick?

Reply:
I think there's code that waits for the response for a period of time.  It's mainly about detecting failures.